staging/src/k8s.io/kubectl/pkg/cmd/taint/taint.go

```diff
@@ -219,15 +227,22 @@ func (o TaintOptions) validateFlags() error {
 // Validate checks to the TaintOptions to see if there is sufficient information run the command.
 func (o TaintOptions) Validate() error {
 	resourceType := strings.ToLower(o.resources[0])
-	validResources, isValidResource := []string{"node", "nodes"}, false
-	for _, validResource := range validResources {
-		if resourceType == validResource {
-			isValidResource = true
-			break
+	fullySpecifiedGVR, _, err := explain.SplitAndParseResourceRequest(resourceType, o.Mapper)
+	if err != nil {
+		return err
+	}
+
+	gvk, _ := o.Mapper.KindFor(fullySpecifiedGVR)
+	if gvk.Empty() {
+		gvk, err = o.Mapper.KindFor(fullySpecifiedGVR.GroupResource().WithVersion(""))
```

Comment:
You don't need that fallback, it will either match or not so you should check error from `KindFor` execution.

Reply:
Done.